org.jacoco.core.test/src/org/jacoco/core/internal/analysis/MethodAnalyzerTest.java

```diff
@@ -57,47 +57,83 @@ public int nextId() {
 		return nextProbeId++;
 	}
 
-	// === Scenario: linear Sequence without branches ===
+	// === Scenario: linear Sequence with and without ignore filtering ===
 
 	private void createLinearSequence() {
 		final Label l0 = new Label();
 		method.visitLabel(l0);
 		method.visitLineNumber(1001, l0);
 		method.visitInsn(Opcodes.NOP);
+		method.visitInsn(Opcodes.NOP);
 		final Label l1 = new Label();
 		method.visitLabel(l1);
 		method.visitLineNumber(1002, l1);
 		method.visitInsn(Opcodes.RETURN);
 	}
 
 	@Test
-	public void testLinearSequenceNotCovered1() {
+	public void linear_instruction_sequence_should_create_1_probe() {
 		createLinearSequence();
 		runMethodAnalzer();
 		assertEquals(1, nextProbeId);
+	}
 
-		assertLine(1001, 1, 0, 0, 0);
+	@Test
+	public void linear_instruction_sequence_should_show_missed_when_no_probe_is_executed() {
+		createLinearSequence();
+		runMethodAnalzer();
+
+		assertLine(1001, 2, 0, 0, 0);
 		assertLine(1002, 1, 0, 0, 0);
 	}
 
 	@Test
-	public void testLinearSequenceNotCovered2() {
+	public void linear_instruction_sequence_should_show_missed_when_probearray_is_null() {
 		createLinearSequence();
 		probes = null;
 		runMethodAnalzer();
-		assertEquals(1, nextProbeId);
 
-		assertLine(1001, 1, 0, 0, 0);
+		assertLine(1001, 2, 0, 0, 0);
 		assertLine(1002, 1, 0, 0, 0);
 	}
 
 	@Test
-	public void testLinearSequenceCovered() {
+	public void linear_instruction_sequence_should_show_covered_when_probe_is_executed() {
 		createLinearSequence();
 		probes[0] = true;
 		runMethodAnalzer();
 
-		assertLine(1001, 0, 1, 0, 0);
+		assertLine(1001, 0, 2, 0, 0);
+		assertLine(1002, 0, 1, 0, 0);
+	}
+
+	/** Filters consecutive NOP instructions as ignored */
+	private static final IFilter NOP_FILTER = new IFilter() {
+		public void filter(final String className, final String superClassName,
+				final MethodNode methodNode, final IFilterOutput output) {
+			AbstractInsnNode node = methodNode.instructions.getFirst();
+			while (node != null) {
```

Comment:
IMO tests should require as little time as possible on understanding, so I would prefer to make this filter a way simpler.

Reply:
@Godin Agreed. The target was to have a filter which filters at least two instructions (to specify a range). Any other ideas for a simplified filter? E.g. just by hard-coded index?

Reply:
@marchof two hard-coded indexes seems ok - this is already the case in this test in `TRY_FINALLY_FILTER`.

Reply:
@Godin Simplifed filter with hardcoded instruction positions.